src/solver/conv_direct_naive_conv.cpp

```diff
@@ -108,7 +110,15 @@ bool IsOutputInt32(const ProblemDescription& problem)
 std::string ConvDirectNaiveConvKernelName(const ProblemDescription& problem)
 {
     std::ostringstream kernel_name;
-    kernel_name << "naive_conv_";
+    if(miopen::IsEnvvarValueEnabled("MIOPEN_USE_PACKED_CONV_REF_KERNEL"))
+    {
+        kernel_name << "naive_conv_packed_";
```

Comment:
Fair point. I want to keep the `packed` variants of kernels for debugging purposes as we found out that debugging becomes easier if I can use the `packed` variant with an `env variable` and quickly establish that the `non-packed` variant or how it is invoked is the source of error. 

Here's my suggestion. I will remove the `packed` variants and this `env` variable in a follow up PR. For this PR, I can put the `if(env var enabled)` part in a `#ifndef NDEBUG` so that it is enabled in debug mode only.

Reply:
To add: use of strides is new in MIOpen it seems. Many kernels still don't support strides and I have found that often tests don't setup the strides correctly, so the env variable helps ease debugging such cases.

Reply:
@amberhassaan 
> I have found that often tests don't setup the strides correctly

This shouldn't be so. Please include fixes for such cases or open a ticket at least. Thanks.

@averinevg Please notice this review thread.

Reply:
@amberhassaan 

> Here's my suggestion. I will remove the `packed` variants and this `env` variable in a follow up PR. For this PR, I can put the `if(env var enabled)` part in a `#ifndef NDEBUG` so that it is enabled in debug mode only.

Maybe we can use "old good" packed variants for convolutions that use packed tensors? Because they are well-tested and may give better performance? What do you think?

Reply:
That is a possibility however, the change is more involved (need logic inside Invoker lambda to test tensors for being packed and non-packed and then call one of two kernels) and I am not sure how much performance benefit we would get (if you look at the kernel code, we replace some integer multiplications with access to kernel parameters, i.e. the strides, which may be cached and cheap to access).

Reply:
@amberhassaan 
> need logic inside Invoker lambda to test tensors for being packed and non-packed and then call one of two kernels

Of course we do not want any additional logic in the invoker (and actually more because in this case we have to build both kernels even if only one will be used). But we can perform the test and select proper kernel outside of InvokerFactory, right in GetSolution(), because packed and non-packed convolution problems will have different Invokers.

Reply:
Next commit will put `packed` name + env variable check in `#ifndef NDEBUG` and a future PR will remove packed kernels altogether. (haven't pushed it yet because I want the CI to run fully).

Reply:
If you would like to get rid of packed kernels, then please make sure that the non-packed kernels are thoroughly tested (several thousands of configs) with packed convolutions. Why: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2334#discussion_r1339255188

@JehandadKhan Do we care about performance of naive kernels?